src/errorBoundary.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { ReactNode } from "react";
+
+interface ErrorBoundaryProps {
+  children: ReactNode;
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+  error: { message: string; stack: string };
+}
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false, error: { message: "", stack: "" } };
+  }
+
+  static getDerivedStateFromError(error: ErrorBoundaryState["error"]): ErrorBoundaryState {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, error };
+  }
+
+  render(): ReactNode {
+    if (this.state.hasError) {
+      // You can render any custom fallback UI
+      return (
+        <div>
+          <h1 className="text-xl font-bold mb-2">Document could not be rendered</h1>
```

Comment:
keep the message generic as error can be anything

Reply:
isnt this generic enough, cos the only reason this will show is if it cannot be rendered. the actual error message is displayed along with this

Reply:
would `Something went wrong with template renderer` be better? not necessary document